litestar/utils/predicates.py

```diff
@@ -306,3 +325,24 @@ def is_pydantic_constrained_field(
         )
     except ImportError:
         return False
+
+
+def is_class_var(annotation: Any) -> bool:
```

Comment:
I'd love for something like this to be as simple as `ParsedType.from_annotation(annotation).origin is ClassVar` or similar.

Following on from discussion in #1447 - there's a bit more going on here with the forward ref handling etc, no tests that demonstrate behavior, and scant docs - all which make it hard to use as an off-the-shelf utility. I'm definitely not trying to say the behavior isn't correct, desired or necessary, I'm just trying to understand why, for example:

- What cases are we trying to exclude by checking for the `__class__` attribute?
- When does `annotation` have and not have a `_name` attribute?
- When are we expecting `get_type_hints()` to return forward references, and should we try to handle them recursively first?

On that last question, if not, why? For example, I know that this can occur if `__future__.annotations` are imported _and_ the annotation is a literal string, but should we make an effort to recursively resolve that inner forward ref? Then, if the name isn't available to be resolved, we can either choose to make that an error (think for parsing reserved kwargs out of handler sigs), or not (maybe such as this case, when parsing arbitrary user models - but at least we know in this case that the `ClassVar` should always be able to be recursively resolved as we always include the `typing` module in our namespace for resolution).

The code that calls this function in `partial.py` calls `get_type_hints(item)` (i.e., without `include_extras=True`) - so any `Annotated` are stripped from the types before calling. If that should always be the case, to aid in making this function more generally available, this should at least be documented somewhere. This test fails, for example:

```py
def test_is_class_var_with_annotated() -> None:
    assert is_class_var(Annotated[ClassVar[int], "meta"]) is True
```

Reply:
addressed